torch/nn/modules/module.py

```diff
@@ -1384,7 +1358,20 @@ def state_dict(self, *args, destination=None, prefix='', keep_vars=False):
                 " and ".join(warn_msg) + " are deprecated. nn.Module.state_dict will not accept them in the future. "
                 "Refer to https://pytorch.org/docs/master/generated/torch.nn.Module.html#torch.nn.Module.state_dict for details.")
 
-        return self._state_dict_impl(destination, prefix, keep_vars)
+        destination._metadata[prefix[:-1]] = local_metadata = dict(version=self._version)
+        self._save_to_state_dict(destination, prefix, keep_vars)
+        for name, module in self._modules.items():
+            if module is not None:
+                sd = module.state_dict(prefix=prefix + name + '.', keep_vars=keep_vars)
```

Comment:
I think this is a bit slower and more memory intensive than just mutating `destination` directly. The difference might be negligible, but possibly not for certain types of modules.